src/blob-uploader.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+'use strict'
+
+const privateVars = new WeakMap()
+
+function blobUploader (apiUrl /* :string */) {
+  if (!apiUrl) {
+    throw new TypeError('blobUploader expects a api URL during instantiation')
+  }
+  privateVars.set(this, {
+    uri: apiUrl
+  })
+}
+
+blobUploader.prototype.uploadBlob = function (
+  blob /*: Blob */
+) /* :Promise<number> */ {
+  if (!blob) {
+    return Promise.reject(new Error('blob argument not provided'))
+  }
+
+  const vars = privateVars.get(this)
+  if (!vars || !vars.hasOwnProperty('uri')) {
+    return Promise.reject(new Error('blobUploader uri not configured'))
+  }
+
+  const request = new Request(vars.uri, {
+    method: 'POST',
+    mode: 'cors'
+  })
+
+  return fetch(request)
+    .then((response) => {
+      if (!response.ok) {
+        return Promise.reject(new Error('Error calling blob api service: ' + response.status + ' ' + response.statusText))
```

Comment:
This error will be caught in the catch below, meaning the error that will be returned to the caller of this function will be `Error calling blob api service: Error calling blob api service: 404 Not Found`

Instead i would reject here with `new Error(response.status + ' ' + response.statusText)`

Same applies to other places rejecting based on `response.ok`